algorithms/gr-bfsrh/cpp/BFS.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+class Graph {
+  int numVertices;
+  list<int>* adjLists;
+  bool* visited;

```

Comment:
Codacy found an issue: [Class 'Graph' is unsafe, 'Graph::visited' can leak by wrong usage.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6318755)